plugins/datastreamer_plugin/fastdds/Participant.cpp

```diff
@@ -387,24 +389,49 @@ void Participant::refresh_types_registered_()
         }
         else
         {
-            // Type not check as registered, check if it is registered in participant
-            if (is_type_registered_(topic_entry.second.first))
+            // If already registered, add it to database and notify listener by own topic discovery callback
+            if (is_type_registered_in_participant_(topic_entry.second.first))
```

Comment:
Same as before, for safer, more mainteinable, and less obscure code: `second` / `first` accessors -> `get<ExpressiveType1>( get<ExpressiveType2>(entry)` ).
As before, consider structured bindings in for looping to simplify accessing